src/DotLiquid/Util/NumericConverter.cs

```diff
@@ -0,0 +1,165 @@
+        {
+            object result = defaultValue;
+            if (value != null)
+            {
+                object convertedObject = value.CoerceToNumericType(formatProvider, defaultValue);
+                if (convertedObject is decimal convertedDecimal)
+                {
+                    result = convertedDecimal;
+                }
+                else if (convertedObject is double convertedDouble)
+                {
+                    result = convertedDouble;
+                }
+                else if (convertedObject is float convertedFloat)
+                {
+                    // https://docs.microsoft.com/en-us/dotnet/standard/base-types/conversion-tables
+                    // Float be widened to double
+                    result = Convert.ToDouble(convertedFloat);
+                }
+                else
+                {
+                    // https://docs.microsoft.com/en-us/dotnet/standard/base-types/conversion-tables
+                    // All integer types can be widened to decimal
+                    result = Convert.ToDecimal(convertedObject);
+                }
+            }
+            return result;
+        }
+
+        /// <summary>
+        /// Coerce an object into a numeric type.
+        /// </summary>
+        /// <param name="value">The string to coerce.</param>
+        /// <param name="formatProvider">The format provider for converting floating point numbers.</param>
+        /// <param name="defaultValue">The value to return if coercion fails.</param>
+        /// <returns>The coerced value as int, long, double or decimal type, or <paramref name="defaultValue"/> if coercion fails.</returns>
+        public static object CoerceToNumericType(this object value, IFormatProvider formatProvider, object defaultValue)
+        {
+            object result = defaultValue;
+            if (value != null)
+            {
+                bool converted = value.TryCoerceToNumericType(formatProvider, out object convertedValue);
+                if (converted)
+                {
+                    result = convertedValue;
+                }
+            }
+            return result;
+        }
+
+        /// <summary>
+        /// Try to coerce an object into a numeric type.
+        /// </summary>
+        /// <param name="value">The string to parse.</param>
+        /// <param name="formatProvider">The format provider for converting floating point numbers.</param>
+        /// <param name="convertedValue">The coerced value as int, long, double or decimal type, or null if parsing fails.</param>
+        /// <returns>true if parsing was successful; Otherwise, false.</returns>
+        public static bool TryCoerceToNumericType(this object value, IFormatProvider formatProvider, out object convertedValue)
+        {
+            if (value != null)
+            {
+                if (IsNumeric(value))

```

Comment:
```suggestion
                if (NumericConverter.IsNumeric(value))
```
Qualify static method calls with the class name so it is easier to distinguish between instance method calls (this.FooBar) and static method calls (Class.FooBar) in a systematic manner.

Reply:
The code base is pretty inconsistent with this, but I've tried to update this patch.